### PR TITLE
Fix null reference in EncryptionProcessor.

### PR DESCRIPTION
**Pull request by mkrueger**
Happens when encrypting/decrypting empty arrays.

Fixes https://github.com/Azure/azure-sdk-for-java/issues/34821

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java [hunk 1]
```diff
@@ -415,7 +415,13 @@ public void encryptAndSerializeProperty(EncryptionSettings encryptionSettings, J
             }
         } else if (propertyValueHolder.isArray()) {
             ArrayNode arrayNode = (ArrayNode) propertyValueHolder;
-            if (arrayNode.elements().next().isObject() || arrayNode.elements().next().isArray()) {
+            Iterator<JsonNode> checkArrayIterator = arrayNode.elements();
+            Boolean hasElements = checkArrayIterator.hasNext();
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java:419]: can be primitive type -> `boolean` ?

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java [hunk 2]
```diff
@@ -564,7 +570,13 @@ public void decryptAndSerializeProperty(EncryptionSettings encryptionSettings, J
...
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java:574]: can be primitive type -> `boolean` ?

**Review by kushagraThapar — APPROVED** [review]
Thanks for the fix @mkrueger , looks good except for one minor comment.

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java]: Fixed

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java]: Fixed

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - Thanks

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - Thanks

**Comment by FabianMeiswinkel** [comment]
CI failures due to known issue with CI pipeline

**Pull request merged**